doc/python/developer.rst

```diff
@@ -62,4 +62,66 @@ And that's it! The check run in the CI is equivalent to here, so you should
 always be able to reproduce failures. And if you install the pre-commit hook,
 you can largely prevent them by always catching them before commit.
 
+Writing Python Commands
+-----------------------
+
+The Flux command design is done in a way that some commands are generated
+from the C code, and others are helpers that come from Python. 
+If you need to add a new command (and want it to show up) there are some tricks you need to know.
+This short guide will help you on this journey! 🏔️
+
+Locations
+~~~~~~~~~
+
+Flux commands can be found under ``src/cmd``. You'll notice many files with
+a prefix "flux-". By default, any file with this prefix will be discovered
+and available to the user as a flux command. As an example, ``flux-run.py`` will
+be available as ``flux run`` and ``flux-job.c`` will be available as ``flux job``. 
+This shows that we have a mix of C derived and Python derived commands. if you are
+interested in how this works, look at ``src/cmd/cmdhelp.c``. If you provide a file with 
+a different name here (e.g., ``supporting-file.py`` it will not be rendered to a command,
+so feel free to do that if the design affords that.
```

Comment:
Didn't we run into problems doing this with your base.py in the "bubble up" PR?  Should from "so feel free..." to end of sentence be dropped?

Reply:
oh right, let me remove that (totally forgot about it... as of one day ago... lol).

Reply:
Feel free to set merge when passing when you're happy with it.